cool-core/src/main/java/com/nus/cool/core/cohort/refactor/storage/CohortWSStr.java

```diff
@@ -0,0 +1,103 @@
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.nus.cool.core.cohort.refactor.storage;
+
+import com.nus.cool.core.io.DataOutputBuffer;
+import com.nus.cool.core.io.Output;
+import com.nus.cool.core.io.compression.Histogram;
+import com.nus.cool.core.io.compression.OutputCompressor;
+import com.nus.cool.core.schema.CompressType;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+
+/**
+ * CohortWS writes the list of users found by a cohort selection query to a file
+ * COOL store the file under the version of cube queried.
+ */
+public class CohortWSStr implements Output {
+
+  /**
+   * A list of user id as keys in string.
+   */
+  private final HashSet<String> usersStrSet = new HashSet<>();
+
+  /**
+   * charset is set in table.yaml and is utf-8 by default.
+   */
+  Charset charset;
+
+  public CohortWSStr(Charset charset) {
+    this.charset = charset;
+  }
+
+  public CohortWSStr() {
+    this.charset = StandardCharsets.UTF_8;
+  }
+
+  /**
+   * Add user id to local list.
```

Comment:
also clarify here 
why load strings and change them to global ids.

Reply:
We store the user id in string type because the global id is managed independently across Cublet. Storing globalID will bring some inconvenience in query processing.